templates/dag-deploy/dag-server-statefulset.yaml

```diff
@@ -114,14 +124,51 @@ spec:
             name: nginx-sidecar-conf
             subPath: nginx.conf
 {{- end }}
+{{- if .Values.loggingSidecar.enabled }}
+        - name: sidecar-log-consumer
+          image: "{{ .Values.loggingSidecar.image }}"
+          env:
+          - name: VECTOR_CONFIG
+            value: /etc/sidecar-logging/vector-config.yaml
+          - name: SIDECAR_LOGS
+            value: /var/log/sidecar-logging-consumer
+          - name: COMPONENT
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.labels['component']
+          - name: RELEASE
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.labels['release']
+          - name: WORKSPACE
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.labels['workspace']
+          volumeMounts:
+          - mountPath: /etc/sidecar-logging
+            name: config-volume
+          - name: sidecar-logging-consumer
+            mountPath: /var/log/sidecar-logging-consumer
+            readOnly: False
+{{- end }}
```

Comment:
@danielhoherd we have this entire section similarity in git sync as well - if we render from helper templates we can reduce code repetition

Reply:
@pgvishnuram yeah, if we can reduce code duplication without complicating the templates then let's do it. This seems like something that should be identical everywhere.